### PR TITLE
Fix quotes and escape characters with WriteQuoteMode.ALL

### PR DESCRIPTION
**Pull request by Floern**
Fixes https://github.com/doyaaaaaken/kotlin-csv/issues/62

### CODE REVIEW

File: src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt [hunk 1]
```diff
@@ -22,6 +22,7 @@ class CsvFileWriter internal constructor(
      */
     private val stateHandler = CsvWriterStateHandler()
 
+    private val quoteNeededChars = setOf('\r', '\n', ctx.quote.char, ctx.delimiter)
```
Comment (doyaaaaaken) [inline src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt:25]: nit:
`quoteNeededChars` is only used on `attachQuote` method, so declare it inside the method.

File: src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriterTest.kt [hunk 2]
```diff
@@ -89,6 +90,15 @@ class CsvFileWriterTest : WordSpec() {
...
+            "write escaped field when a field contains quoteChar in it (quote all)" {
+                val rows = listOf(listOf("a", "\"b", "c"), listOf("d", "e", "f\""))
+                val expected = "\"a\",\"\"\"b\",\"c\"\r\n\"d\",\"e\",\"f\"\"\"\r\n"
+                csvWriter { quote { mode = WriteQuoteMode.ALL } }.open(testFileName) {
+                    writeRows(rows)
+                }
+                val actual = readTestFile()
+                actual shouldBe expected
+            }
             "write escaped field when a field contains delimiter in it" {
```
Comment (doyaaaaaken) [inline src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriterTest.kt:102]: I'd be happy to write additional test code written in below comment.
https://github.com/doyaaaaaken/kotlin-csv/issues/62#issuecomment-696770119

(Maybe `Customized CsvWriter` section is proper. 
https://github.com/doyaaaaaken/kotlin-csv/blob/ae8721271f805303118fe8acc90516d7565a2290/src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterTest.kt#L125 )

**Review by doyaaaaaken — COMMENTED** [review]
Thank you so much for your pull requests! 🙇 
Just a few minor improvements, please.

File: src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt (hunk 1 above)
Comment (Floern) [reply src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt:25]: It doesn't seem very efficient to me to create a new set every time `attachQuote()` is called, given that the contents of the set don't change. Is this intended?

File: src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriterTest.kt (hunk 2 above)
Comment (Floern) [reply src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriterTest.kt:102]: Ah, missed that test section. Will change that 👍

File: src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt (hunk 1 above)
Comment (doyaaaaaken) [reply src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt:25]: I understand your point. I agree with your! 👍

File: src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriterTest.kt (hunk 2 above)
Comment (Floern) [reply src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriterTest.kt]: I added your test to the "Customized CsvWriter" section.

**Comment by doyaaaaaken** [comment]
Great! Thanks!! 🎉

**Pull request merged**